### PR TITLE
#13 adding created information to nodes.

### PR DESCRIPTION
**Pull request by kcnklub**
LFGGGGGGGGGGGGGGGGGGGGGGGG

### CODE REVIEW

File: src/main/java/food/finder/foodfinder/FoodFinderApplication.java [hunk 1]
```diff
@@ -10,4 +10,4 @@
 	public static void main(String[] args) {
 		SpringApplication.run(FoodFinderApplication.class, args);
 	}
-}
\ No newline at end of file
+}
```
Comment (AndrewSpeelman) [inline src/main/java/food/finder/foodfinder/FoodFinderApplication.java:13]: Why did this pick up on the diff? Whitespace?

File: src/main/java/food/finder/foodfinder/user/UserController.java [hunk 2]
```diff
@@ -7,6 +7,8 @@
 import org.springframework.web.bind.annotation.RestController;
 import reactor.core.publisher.Mono;
 
+import java.util.Date;
+
 @RequiredArgsConstructor
 @RestController
```
Comment (AndrewSpeelman) [inline src/main/java/food/finder/foodfinder/user/UserController.java:13]: Woahhhh this is really cool

File: src/main/java/food/finder/foodfinder/user/User.java [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import org.springframework.data.neo4j.core.schema.Property;
 import org.springframework.data.neo4j.core.schema.Relationship;
 
+import java.util.Date;
 import java.util.HashSet;
 import java.util.Set;
```
Comment (AndrewSpeelman) [inline src/main/java/food/finder/foodfinder/user/User.java:14]: Apparently, you can't line comment on something that doesn't have a diff ...

lombok is really cool, so all the fields in this class now have get and set methods. is there any downside to using lombok in a project?

File: src/main/java/food/finder/foodfinder/user/UserController.java [hunk 4]
```diff
@@ -18,6 +20,11 @@
     @PostMapping("/createNewUser")
     Mono<User> createNewUser(@RequestBody User user) {
        return userService.makeSureNewUserDoesNotAlreadyExist(user.getEmail())
-               .then(userRepository.save(user));
+               .then(saveUser(user));
+    }
+
+    Mono<User> saveUser(User user) {
```
Comment (AndrewSpeelman) [inline src/main/java/food/finder/foodfinder/user/UserController.java:26]: I'm going ot have to do more research on Mono, quick google got me a lil mind boggled

File: src/main/java/food/finder/foodfinder/user/User.java [hunk 5]
```diff
@@ -22,11 +23,14 @@
...
+    @Property
+    private Date createDate;
+
     @Relationship(type = "LIKED", direction = Relationship.Direction.OUTGOING)
     private Set<Place> likedPlaces = new HashSet<>();
 
     public User(String email, String username) {
         this.email = email;
         this.username = username;
     }
```
Comment (AndrewSpeelman) [inline src/main/java/food/finder/foodfinder/user/User.java:35]: Should createdDate be defaulted here?

**Review by AndrewSpeelman — COMMENTED** [review]
github is really weird XD

File: src/main/java/food/finder/foodfinder/user/User.java (hunk 3 above)
Comment (kcnklub) [reply src/main/java/food/finder/foodfinder/user/User.java:14]: The only downside that I can think of is having more methods are are used but that happens at compile time so doesn't effect the readability. There are some caveats using Lombok in combination with other libraries like hibernates but they are really specific cases.

File: src/main/java/food/finder/foodfinder/FoodFinderApplication.java (hunk 1 above)
Comment (kcnklub) [reply src/main/java/food/finder/foodfinder/FoodFinderApplication.java:13]: I was confused at that too

File: src/main/java/food/finder/foodfinder/user/User.java (hunk 5 above)
Comment (kcnklub) [reply src/main/java/food/finder/foodfinder/user/User.java:35]: Good question I think that it ca. be null but can't be saved if null.

**Comment by kcnklub** [comment]
> github is really weird XD 

That is the truth takes a bit to get used to but this is what everyone else uses.

**Comment by kcnklub** [comment]
All of this was don't via my phone btw.

**Pull request merged**